lib/astarte_vmq_plugin/config.ex

```diff
@@ -67,6 +68,44 @@ defmodule Astarte.VMQ.Plugin.Config do
     Application.put_env(:astarte_rpc, :amqp_connection, astarte_rpc_amqp_opts)
   end
 
+  defp init_ssl_options(amqp_options) do
+    amqp_ssl = Application.get_env(:astarte_vmq_plugin, :amqp_ssl, [])
+    ssl_options = build_ssl_options(amqp_ssl, amqp_options)
+    Keyword.put(amqp_options, :ssl_options, ssl_options)
+  end
+
+  defp build_ssl_options([], amqp_options), do: amqp_options
+
+  defp build_ssl_options(amqp_ssl, amqp_options) do
+    case Keyword.fetch(amqp_ssl, :disable_sni) do
+      {:ok, false} ->
+        populate_ssl_options(amqp_ssl, amqp_options)
+
+      _ ->
+        ssl_disable_sni(amqp_options)
+    end
+  end
+
+  defp populate_ssl_options(amqp_ssl, amqp_options) do
+    server_name_indication =
+      case Keyword.fetch(amqp_ssl, :custom_sni) do
+        {:ok, custom_sni} ->
+          custom_sni
+
+        _ ->
+          Keyword.fetch!(amqp_options, :host)
+      end
+
+    Keyword.fetch!(amqp_options, :ssl_options)
+    |> Keyword.put(:server_name_indication, to_charlist(server_name_indication))
+    |> Keyword.put(:verify, :verify_peer)
+  end
+
+  defp ssl_disable_sni(amqp_options) do
+    Keyword.fetch!(amqp_options, :ssl_options)
+    |> Keyword.put(:server_name_indication, :disable)
+  end
+
```

Comment:
Almost there, but there are still a couple of problems:
- You are duplicating the code that merges the options, and in doing so you risk forgetting stuff (for example you add `verify_peer` only in one of the two. `build_ssl_options` should just build the additional `ssl_options` and the merge should happen `init_ssl_options`, avoiding duplication.
- You should initialize the common options at the beginning (i.e. `:verify_peer`) and the other functions should probably be named `populate_sni` or something like that, since they only handle the SNI part.